src/Network/NWEthernetChannel.cs

```diff
@@ -76,6 +78,39 @@ public NWEthernetChannel (ushort ethernetType, NWInterface networkInterface)
 			InitializeHandle (nw_ethernet_channel_create (ethernetType, networkInterface.Handle));
 		}
 
+#if NET
+		[SupportedOSPlatform ("macos13.0")]
+		[UnsupportedOSPlatform ("tvos")]
+		[UnsupportedOSPlatform ("ios")]
+#else
+		[NoWatch]
+		[NoTV]
+		[NoiOS]
+		[Mac (13,0)]
+#endif
+		[DllImport (Constants.NetworkLibrary)]
+		static extern OS_nw_ethernet_channel nw_ethernet_channel_create_with_parameters (ushort ether_type, OS_nw_interface @interface, OS_nw_parameters parameters);
+
+#if NET
+		[SupportedOSPlatform ("macos13.0")]
+		[UnsupportedOSPlatform ("tvos")]
+		[UnsupportedOSPlatform ("ios")]
+#else
+		[NoWatch]
+		[NoTV]
+		[NoiOS]
+		[Mac (13,0)]
+#endif
+		public NWEthernetChannel (ushort ethernetType, NWInterface networkInterface, NWParameters parameters) {
+			if (networkInterface is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (networkInterface));
+
+			if (parameters is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (parameters));
+
+			InitializeHandle (nw_ethernet_channel_create_with_parameters (ethernetType, networkInterface.Handle, parameters.Handle));
```

Comment:
You can use GetNonNullHandle here instead of manually checking above.

Reply:
But I need to check nevertheless to throw the exception.

Reply:
No, `GetNonNullHandle` throws the exception:

https://github.com/xamarin/xamarin-macios/blob/75112818b6d423b1d474acb14e8c33147d33471a/src/ObjCRuntime/INativeObject.cs#L30-L33